.github/CONTRIBUTING.md

```diff
@@ -91,7 +78,53 @@ and to [have PowerShell installed](https://github.com/PowerShell/PowerShell#get-
+
+After running this script you should validate that the update didn't introduce any breaking changes; see the next section for how to do this.
+
+
+### Validating your changes
+
+When you contribute any code to the Katas, you need to validate that everything works the way it is supposed to work. Here are the key points to check (they might or might not be applicable to your change, depending on what you modified):
+
+1. **Local development**  
+   1. Check that the kata/tutorial you modified builds using `dotnet build` (if you modified the files in the project).
+   2. Check that the notebook version of the kata/tutorial opens using `jupyter notebook` (if you modified the notebook file).
+   3. Check that the reference solutions for the tasks pass the tests.  
+      You can use Jupyter Notebook front-end of the kata you're working on to validate this (i.e., to check that all tasks have correct reference solutions for them, and that all tests used in the notebook actually exist in the project).  
+      
+      To validate the kata, use the [`scripts/validate-notebooks.ps1`](../scripts/validate-notebooks.ps1) script. 
+      For example, to validate BasicGates kata run the following command from the PowerShell prompt from the root directory of the QuantumKatas project:
+
+      ```powershell
+         PS> ./scripts/validate-notebooks.ps1 ./BasicGates/BasicGates.ipynb
+      ```
+
+      To use this script, you need to be able to [run Q# Jupyter notebooks locally](https://docs.microsoft.com/quantum/install-guide/qjupyter) 
+and to [have PowerShell installed](https://github.com/PowerShell/PowerShell#get-powershell).
+
+   4. If you do a bulk update of the katas, testing each of them individually will take too much time; you can streamline the testing using the scripts used by our continuous integration. 
+   It is also a good idea to check a representative kata (we recommend [Measurements](https://github.com/microsoft/QuantumKatas/tree/master/Measurements)) manually to see if there is any issue not covered by automated checks, such as different error format, a dramatic performance degradation etc.
+
+2. **Running on Binder**  
+   The Katas can be run online on [Binder](https://mybinder.org); when you make a potentially breaking change (such as an update to the new QDK version or modifying any package dependencies), you need to make sure that this still works.  
+   You can check this by pushing your changes to a branch on GitHub and navigating to the Binder link used for the Katas (https://mybinder.org/v2/gh/Microsoft/QuantumKatas/master?filepath=index.ipynb) and change account name (`microsoft`) and branch (`master`) in the url to your Github username and branch name, respectively. After that you can navigate to the kata you want to check using the links from index notebook.
```

Comment:
```suggestion
   You can check this by pushing your changes to a branch on GitHub and navigating to the Binder link used for the Katas (https://mybinder.org/v2/gh/Microsoft/QuantumKatas/master?filepath=index.ipynb) and change account name (`microsoft`) and branch (`master`) in the url to your GitHub username and branch name, respectively. After that you can navigate to the kata you want to check using the links from index notebook.
```

Reply:
Nice catch! I'll note this for the next time I edit this file (around next week)